src/main/scala/ru/org/codingteam/horta/plugins/pet/Pet.scala

```diff
@@ -50,55 +40,58 @@ class Pet(roomId: String, location: ActorRef) extends Actor with ActorLogging {
-          if (pet.random.nextInt(CHANCE_OF_ATTACK) == 0 && coinHolders.size > 0) {
-            (location ? GetParticipants).mapTo[Protocol.ParticipantCollection].map { map =>
-              val possibleVictims = map.keys map ((x: String) => StringUtils.parseResource(x))
-              val victim = pet.randomChoice((coinHolders.toSet & possibleVictims.toSet).toList)
-              coins ! UpdateUserPTCWithOverflow("pet aggressive attack", victim, -ATTACK_PENALTY)
-              credential.map { implicit c =>
-                sayToEveryone(random("%s aggressively attacked %s").format(nickname, victim)
-                  + random(" due to hunger, taking some of his PTC.") + " "
-                  + localize("%s loses %dPTC.").format(ATTACK_PENALTY))
-                FULL_SATIATION
-              }
-            }.flatMap(identity)
-          } else {
-            credential.map { implicit c =>
-              sayToEveryone(random("%s is searching for food.").format(nickname))
-              satiation
+    pet match {
+      case DyingPet() =>
+        PetTickResponse(pet.killed, takeDeathPenalty, { implicit c =>
+          sayToEveryone(random("%s is dead.").format(nickname) + " " +
+          localize("All members have lost %dPTC.").format(DEATH_PENALTY))
+        })
+
+      case HungerPet() if pet.random.nextInt(SPARSENESS_OF_EVENTS) == 0 =>
+        if (pet.random.nextInt(CHANCE_OF_ATTACK) == 0 && coinHolders.size > 0) {
+          val victim = getPetVictimAsync(pet, coinHolders)
+          PetTickResponse(pet.satiated, attackVictim(context.dispatcher, victim), { implicit c =>
+            victim.map { v =>
+              sayToEveryone(random("%s aggressively attacked %s").format(nickname, v)
+                + random(" due to hunger, taking some of his PTC.") + " "
+                + localize("%s loses %dPTC.").format(ATTACK_PENALTY))
             }
-          }
-        } else if (health <= HEALTH_BOUNDS._2 && pet.health > HEALTH_BOUNDS._1) {
-          credential.map { implicit c =>
-            sayToEveryone(random("%s's health is low.").format(nickname))
-            satiation
-          }
+          })
         } else {
-          Future.successful(satiation)
-        }).map(newSatiation => pet.copy(alive = alive, health = health, satiation = newSatiation))
+          PetTickResponse(pet, noAction, { implicit c =>
+            sayToEveryone(random("%s is searching for food.").format(nickname))
+          })
+        }
+
+      case IllPet() =>
+        PetTickResponse(pet, noAction, { implicit c =>
+          sayToEveryone(random("%s's health is low.").format(nickname))
+        })
+
+      case _ => PetTickResponse(pet, noAction, { c => })
+    }
+  }
+
+  private def getPetVictimAsync(pet: PetData, coinHolders: Iterable[String]) = {
+    (location ? GetParticipants).mapTo[Protocol.ParticipantCollection].map { map =>
+      val possibleVictims = map.keys map ((x: String) => StringUtils.parseResource(x))
```

Comment:
Is it possible to replace `(x: String) => StringUtils.parseResource(x)` with something like `StringUtils.parseResource(_)`?